core/print_api.php

```diff
@@ -1894,16 +1894,19 @@ function print_bug_attachments_list( $p_bug_id, $p_security_token ) {
  * a valid security token, previously generated by form_security_token().
  * Use this to avoid performance issues when loading pages having many calls to
  * this function, such as print_bug_attachments_list().
- * @param array $p_attachment An attachment array from within the array returned by the file_get_visible_attachments() function.
+ * @param array $p_attachment An attachment array from within the array returned
+ *                            by the file_get_visible_attachments() function.
  * @param string $p_security_token The security token to use for deleting attachments.
  * @return void
  */
 function print_bug_attachment( array $p_attachment, $p_security_token ) {
-	$t_show_attachment_preview = $p_attachment['preview'] && $p_attachment['exists'] && ( $p_attachment['type'] == 'text' || $p_attachment['type'] == 'image' );
+	$t_show_attachment_preview = $p_attachment['preview']
+		&& $p_attachment['exists']
+		&& ( $p_attachment['type'] == 'text' || $p_attachment['type'] == 'image' );
```

Comment:
The check for exists, can download, size, text or image are all done as part of setting the 'preview' flag.